arbos/addressSet/addressSet.go

```diff
@@ -84,6 +86,44 @@ func (aset *AddressSet) AllMembers(maxNumToReturn uint64) ([]common.Address, err
 	return ret, nil
 }
 
+func (aset *AddressSet) ClearList() error {
+	size, err := aset.size.Get()
+	if err != nil || size == 0 {
+		return err
+	}
+	for i := uint64(1); i <= size; i++ {
+		err = aset.backingStorage.ClearByUint64(i)
+		if err != nil {
+			return err
+		}
+	}
+	return aset.size.Clear()
+}
+
+func (aset *AddressSet) RectifyMapping(addr common.Address) error {
+	isOwner, err := aset.IsMember(addr)
+	if !isOwner || err != nil {
+		return errors.New("RectifyMapping: Address is not an owner")
+	}
+
+	// If the mapping is correct, RectifyMapping shouldnt do anything
+	// Additional safety check to avoid corrupting the mapping after the initial fix
+	addrAsHash := common.BytesToHash(addr.Bytes())
+	slot, _ := aset.byAddress.GetUint64(addrAsHash)
+	atSlot, _ := aset.backingStorage.GetByUint64(slot)
+	if atSlot == addrAsHash && atSlot != common.BytesToHash(common.Address{}.Bytes()) {
```

Comment:
I think adding the check is a bit safer, both in case the zero address was in the set, and in case as you mentioned the cleanup doesn't work properly. I don't think it's necessary for any existing chains, but I think it's better to cover all the edge cases here.

Reply:
cool.